utils/jsonSigning.js

```diff
@@ -0,0 +1,75 @@
+// JSON Cleartext Signing
+'use strict'
+
+const rsasign = require('jsrsasign') // eslint-disable-line no-unused-vars
+const ServerError = require('../errors/server-error')
+const _ = require('lodash')
+
+// Sign JSON object, using ECDSA P-256 private key and public key.
+// Insert 'signature' block in the JSON object.
+// Warning: there must not be an existing 'signature' block.
+module.exports.sign = function (json, prvKey, pubKey) {
+  if (!prvKey) throw new ServerError('Problem reading private key for JSON signing')
+  if (!pubKey) throw new ServerError('Problem reading public key for JSON signing')
+  const strJWS = KJUR.jws.JWS.sign('ES256', // eslint-disable-line no-undef
+                                   JSON.stringify({alg: 'ES256'}),
+                                   JSON.stringify(json),
+                                   prvKey)
+  const jwsArray = strJWS.split('.')
+  // Get x and y coordinates
+  const pubKeyHex = pubKey.pubKeyHex
+  // According to jsrsasign documentation,
+  // pubkey hex length should be 2 (header) + 64 (X) + 64 (Y) = 130.
+  // first two chracters must be "04".
+  if (pubKeyHex.length !== 130 || pubKeyHex.indexOf('04') !== 0) {
+    throw new ServerError('Problem decoding public key for JSON signing')
+  }
+  const hX = pubKeyHex.slice(2, 66)
+  const hY = pubKeyHex.slice(66, 130)
+
+  const signedJSON = _.cloneDeep(json)
+  signedJSON.signature = {
+    'algorithm': 'ES256',
+    'publicKey': {
+      'type': 'EC',
+      'curve': 'P-256',
+      'x': hX,
+      'y': hY
+    },
+    'value': jwsArray[2]
+  }
+  return signedJSON
+}
+
+// Verify 'signature' block on the input JSON object has the correct signature value.
+module.exports.verify = function (json, pubKey) {
+  if (!json || !json.signature || !json.signature.value) {
+    throw new ServerError('Invalid input for JSON verification')
+  }
+  const jsonWithoutSignature = _.cloneDeep(json)
```

Comment:
I think you can use `_.omit(json, 'signature')` here, since I don't think `omit` mutates the underlying object.